### PR TITLE
Task/and 1042/fix presence draw issue And UI Feedback

### PR DESCRIPTION
**Pull request by andela-wakinsanmi**

### CODE REVIEW

File: layer-atlas/src/main/java/com/layer/ui/message/MessageItemViewModel.java [hunk 1]
```diff
@@ -176,4 +177,13 @@ public void setShouldDisplayAvatarSpace(boolean shouldDisplayAvatarSpace) {
     public boolean getShouldDisplayAvatarSpace() {
         return mShouldDisplayAvatarSpace;
     }
+
+    public void setIsPreviousMessageFromSameUser(boolean isPreviousMessageFromSameUser) {
```
Comment (thecombatwombat) [inline layer-atlas/src/main/java/com/layer/ui/message/MessageItemViewModel.java]: Why is this here?

File: layer-atlas/src/main/res/layout/ui_message_item.xml [hunk 2]
```diff
@@ -88,19 +88,20 @@
                     android:textSize="@dimen/layer_ui_text_size_message_item_metadata" />
             </LinearLayout>
 
+            <!--Space use to separate messages base on Cluster rule-->
             <Space
                 android:id="@+id/cluster_space"
                 android:layout_width="match_parent"
-                android:layout_height="@dimen/layer_ui_message_cluster_space_gap"
+                android:layout_height="@{viewModel.isPreviousMessageFromSameUser ? @dimen/layer_ui_zero_padding : @dimen/layer_ui_message_cluster_space_gap, default=@dimen/layer_ui_message_cluster_space_gap}"
```
Comment (thecombatwombat) [inline layer-atlas/src/main/res/layout/ui_message_item.xml]: Refer to question above. What happens when the previous message is deleted?

File: layer-atlas/src/main/java/com/layer/ui/message/MessageItemViewModel.java (hunk 1 above)
Comment (andela-wakinsanmi) [reply layer-atlas/src/main/java/com/layer/ui/message/MessageItemViewModel.java]: This is to check if the previousMessage is from the same user in order to give it appropriate padding? Before, we use the the time stamp to space the messageItem, now we want to group message from the same user together.

File: layer-atlas/src/main/java/com/layer/ui/util/imagecache/PicassoImageCacheWrapper.java [hunk 3]
```diff
@@ -99,23 +99,30 @@ public void resumeTag(String picassoTag) {
...
+                && imageRequestParameters.getResizeHeightTo() >= 0;
+
         if (imageRequestParameters.isShouldCenterImage()) {
             requestCreator = requestCreator.centerCrop();
         }
 
         if (imageRequestParameters.getRotateAngleTo() == 0) {
-            requestCreator.resize(imageRequestParameters.getResizeWidthTo(),
-                    imageRequestParameters.getResizeHeightTo());
+            if (hasValidResizeField) {
```
Comment (thecombatwombat) [inline layer-atlas/src/main/java/com/layer/ui/util/imagecache/PicassoImageCacheWrapper.java]: Why can't this block be simplified to : 

```        
if (imageRequestParameters.getResizeWidthTo() >= 0 && imageRequestParameters.getResizeHeightTo() >= 0) {
    requestCreator.resize(imageRequestParameters.getResizeWidthTo(), imageRequestParameters.getResizeHeightTo());
}

if (imageRequestParameters.getRotateAngleTo() != 0) {
    requestCreator.rotate(imageRequestParameters.getRotateAngleTo());
}
```

**Comment by andela-wakinsanmi** [comment]
This is from the old build screenshot_20170925-224258

And this is from this branch
screenshot_20170925-224048

**Comment by thecombatwombat** [comment]
@andela-wakinsanmi This looks fine to me, but please hold off on merging these changes. 

I'm currently working on some changes and I want to make sure that I port over your changes to my branch correctly. 

Once I'm ready I'll merge this in for you.

**Comment by andela-wakinsanmi** [comment]
Ok

File: layer-atlas/src/main/res/layout/ui_message_item_legacy.xml [hunk 4]
```diff
@@ -77,33 +77,34 @@
...
+                    android:textSize="@dimen/layer_ui_text_size_message_item_metadata" />
             </LinearLayout>
 
+            <!--Space use to separate messages base on Cluster rule-->
             <Space
                 android:id="@+id/cluster_space"
                 android:layout_width="match_parent"
-                android:layout_height="@dimen/layer_ui_message_cluster_space_gap"
-                android:visibility="@{viewModel.clusterSpaceVisible ? View.VISIBLE : View.GONE}"/>
+                android:layout_height="@{@dimen/layer_ui_message_cluster_space_gap, default=@dimen/layer_ui_message_cluster_space_gap}"
```
Comment (thecombatwombat) [inline layer-atlas/src/main/res/layout/ui_message_item_legacy.xml]: why is the default the same as the normal argument? why is a default needed here? infact, why is anything databinding related needed here?

Comment (andela-wakinsanmi) [reply layer-atlas/src/main/res/layout/ui_message_item_legacy.xml]: Thanks, I have removed it
